src/pages/add-edit/add-edit-page.jsx

```diff
@@ -67,239 +79,250 @@ export default class AddEditEventPage extends React.Component {
+        : loc.value.trim(),
+    });
+    // Save the dirty version to be passed on to the text field
+    this.setState({ locationRaw: loc.value });
+  };
+
+  allDayToggled = e => this.updateEventDatum({ allDay: e.currentTarget.checked });
+
+  setStart = (start) => {
+    const timeDelta = start.diff(this.state.eventData.start);
+    this.updateEventDatum({
+      start,
+      // Shift the end time to maintain the same duration
+      end: moment(this.state.eventData.end).add(timeDelta, 'ms'),
+    });
+  };
+
+  setEnd = end => this.updateEventDatum({ end });
+
+  doesRecurToggled = e => this.updateEventDatum({
+    recurrence: e.currentTarget.checked
+      ? deepcopy(this.state.defaultRecurrence)
+      : null,
+  });
+
+  recurrenceRuleChanged = recurrence => this.updateEventDatum({ recurrence });
+
+  descriptionChanged = description => this.updateEventDatum({ description });
+
+  labelToggled = (labelName) => {
+    const labels = this.state.eventData.labels.slice(); // Make a copy of the list
+    const labelIndex = labels.indexOf(labelName);
+    if (labelIndex > 0) { // The label is already on the event
+      labels.splice(labelName, 1); // Remove the label
+    } else {
+      labels.push(labelName); // Add the label
+    }
+    this.updateEventDatum({ labels });
+  };
 
-    updateEventDatum = delta => this.setState({ eventData: Object.assign({}, this.state.eventData, delta) });
+  visibilityChanged = visibility => this.updateEventDatum({ visibility });
 
-    render() {
-      if (!this.state.eventData) {
-        return (
-          <div>
-            <h1>Loading...</h1>
-          </div>
-        );
-      }
+  updateEventDatum = delta => this.setState({ eventData: Object.assign({}, this.state.eventData, delta) });
 
-      const editingExisting = this.state.eventData.id || this.state.eventData.sid;
-      const pageTitle = editingExisting ? 'Edit Event' : 'Add Event';
-      const submitButtonText = editingExisting ? 'Update Event' : 'Add Event';
-      const formUrl = 'https://goo.gl/forms/2cqVijokICZ5S20R2';
+  render() {
+    if (!this.state.eventData) {
       return (
-        <div className="row content-container">
-          <span className="content-container">
-            <h1 className="page-title">
-              <MenuIconButton onClick={this.props.toggleSidebarCollapsed} tooltip="Show/Hide Sidebar" />
-              {pageTitle}
-            </h1>
-          </span>
-          <div className="event-info-container">
-            <input
-              id="event-title"
-              type="text"
-              placeholder="Title"
-              className="wide-text-box single-line-text-box medium-text-box"
-              value={this.state.eventData.title}
-              onChange={this.titleChanged}
+        <div>
+          <h1>Loading...</h1>
+        </div>
+      );
+    }
+
+    const editingExisting = this.state.eventData.id || this.state.eventData.sid;
```

Comment:
Note that `editingExisting` has a type `string | undefined`. (This isn't quite the same as [Flow's](https://flow.org/en/docs/types/maybe/) `?string`, because `undefined !== null`.) I think that's okay the way it's used here. To assure it's a boolean, you could use `Boolean(this.state.eventData.id) || Boolean(this.state.eventData.sid)` or `Boolean(this.state.eventData.id || this.state.eventData.sid)`.

Reply:
Yeah, it should be fine.